tox.ini

```diff
@@ -64,6 +64,27 @@ setenv =
     {[base]setenv}
     SYNAPSE_POSTGRES = 1
 
+# A test suite for the oldest supported versions of Python libraries, to catch
+# any uses of APIs not available in them.
+[testenv:py27-old]
+skip_install=True
+deps =
+    # Old automat version for Twisted
+    Automat == 0.3.0
+
+    mock
+    lxml
+commands =
+    /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
+    # Make all greater-thans equals so we test the oldest version of our direct
+    # dependencies, but make the psycopg2 version one which can compile against
+    # PostgreSQL 10 and the pyopenssl one which can work against an OpenSSL 1.1
+    # compiled cryptography.
+    /bin/sh -c 'python -m synapse.python_dependencies | sed -e "s/>=/==/g" -e "s/psycopg2==2.6/psycopg2==2.7/" -e "s/pyopenssl==16.0.0/pyopenssl==17.0.0/" | xargs pip install'
```

Comment:
[given we're not running the tests against postgres here, why do we care about psycopg2 at all?]